tests/org.eclipse.papyrus.uml.diagram.sequence.runtime.tests/src/org/eclipse/papyrus/uml/diagram/sequence/runtime/internal/edit/policies/tests/ExecutionSpecificationCreationEditPolicyUITest.java

```diff
@@ -0,0 +1,69 @@
+/*****************************************************************************
+ * Copyright (c) 2018 EclipseSource and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   EclipseSource - Initial API and implementation
+ *   
+ *****************************************************************************/
+
+package org.eclipse.papyrus.uml.diagram.sequence.runtime.internal.edit.policies.tests;
+
+import static org.eclipse.papyrus.uml.diagram.sequence.runtime.tests.matchers.GEFMatchers.EditParts.isAt;
+import static org.eclipse.papyrus.uml.diagram.sequence.runtime.tests.matchers.GEFMatchers.EditParts.isSized;
+import static org.eclipse.papyrus.uml.diagram.sequence.runtime.tests.rules.EditorFixture.DEFAULT_SIZE;
+import static org.eclipse.papyrus.uml.diagram.sequence.runtime.tests.rules.EditorFixture.at;
+import static org.eclipse.papyrus.uml.interaction.tests.matchers.NumberMatchers.isNear;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.eclipse.gef.EditPart;
+import org.eclipse.papyrus.uml.diagram.sequence.runtime.internal.edit.policies.ExecutionSpecificationCreationEditPolicy;
+import org.eclipse.papyrus.uml.diagram.sequence.runtime.internal.providers.SequenceElementTypes;
+import org.eclipse.papyrus.uml.interaction.tests.rules.ModelResource;
+import org.junit.Test;
+
+/**
+ * Integration tests for {@link ExecutionSpecificationCreationEditPolicy}
+ */
+@SuppressWarnings("restriction")
+@ModelResource("one-exec.di")
+public class ExecutionSpecificationCreationEditPolicyUITest extends AbstractGraphicalEditPolicyUITest {
```

Comment:
Nice! Unfortunately, this test fails on my Mac:

```
Expected: location point that x that 276 ± 1 and y that 170 ± 1
     but: location y was <165>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
	at org.eclipse.papyrus.uml.diagram.sequence.runtime.internal.edit.policies.tests.ExecutionSpecificationCreationEditPolicyUITest.createNestedExecution(ExecutionSpecificationCreationEditPolicyUITest.java:64)
```

On the plus side, all of the other tests still pass (with no new skips). 😉

Reply:
Hmmm, interesting! Again an issue with different platforms. I updated the test so it does get the initial Y value and compare the value after execution with the initial one. 
Funny fact is that this model is also used in other classes, like ExecutionSnappingUITest. In all other classes, it is not used directly, only with offset to set the message connection point, or not used at all ;)
I would still be very happy to understand this difference. The position of the execution is set on the lifeline body, at least in notation. The height of the header is set also, so as its y position. The interaction iself may have a different layout with its label on top, given the font? This is the only element that does not have a size pushed in notation.

Reply:
Now, on my system, the tests are green.  So, that's good.  However, there's now an assumption failure that skips the `ExecutionMoveUITest::moveSyncMessageEnd` test.  It seems that the sync message is no longer located in the diagram where the test expects it to be.